### PR TITLE
feat(source): support temporary source

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Support create temporary sources. If TEMPORARY or TEMP is specified, the source is created as a temporary source. Temporary sources are automatically dropped at the end of a session. 
- The temporary source is expected to be selected by batch queries, not streaming queries. So when it is used by materialized view, the meta node must detect it and reject it.
- The current PR supports temporary source with minimum effort, so we don't care about the database name and schema name, but only care about the source name, i.e. only temporary sources are maintained by a simple mapping (source_name -> source catalog)
- temporary sources can only be shown via `show sources` but not other system tables.

Example: 
```sql

create temporary source s3 (
    id integer ,
    v_varchar varchar,
    v_smallint smallint,
    v_integer integer,
    v_bigint bigint,
    v_float real,
    v_double double precision,
    v_timestamp timestamp,
    v_bytea bytea,
    v_interval interval,
    v_date date,
    v_time time,
    v_timestamptz timestamptz,
    v_jsonb jsonb
) with (
  connector = 'kafka',
  topic = 'test-rw-sink-without-snapshot',
  properties.bootstrap.server = '127.0.0.1:29092',
  scan.startup.mode = 'earliest'
) FORMAT PLAIN ENCODE JSON;
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by tabVersion** [comment]
so fast 🥵

File: e2e_test/source/basic/temporary_kafka_batch.slt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+1 1
+2 22
+3 333
+4 4444
+
+statement error expected format
+select * from s1 where _rw_kafka_timestamp > 'abc'
+
+statement error out of range
+select * from s1 where _rw_kafka_timestamp < TO_TIMESTAMP(2147483647 + 1)
```
Comment (tabVersion) [inline e2e_test/source/basic/temporary_kafka_batch.slt:45]: whats `2147483647` mean

Comment (chenzl25) [reply e2e_test/source/basic/temporary_kafka_batch.slt:45]: Just copy paste from `kafka_batch.slt` which should represents  2038-01-19 05:14:07

**Review by StrikeW — COMMENTED** [review]
Would we expose this syntax to user? Or it is only used for the validation of connectivity to upstream in the Cloud?

**Comment by chenzl25** [comment]
> Would we expose this syntax to user? Or it is only used for the validation of connectivity to upstream in the Cloud? I mean whether we need to document it in the user guide.

We won't expose this syntax to users. It is used by ourselves internally because users can always use a source instead of a temporary source.

**Review by BugenZhao — APPROVED** [review]
Generally LGTM.

Considering that cloud portal has exclusive access to the session where a temporary source is created, despite crude, I think the current implementation is sufficient to work. However, it appears to lack comprehensiveness for user interaction. Maybe we should keep it undocumented and internal-use-only.

File: src/frontend/src/session.rs [hunk 2]
```diff
@@ -652,6 +653,38 @@ pub struct SessionImpl {
     last_idle_instant: Arc<Mutex<Option<Instant>>>,
 
     cursor_manager: Arc<CursorManager>,
+
+    /// temporary sources for the current session
+    temporary_source_manager: Arc<Mutex<TemporarySourceManager>>,
+}
+
+#[derive(Default, Clone)]
+pub struct TemporarySourceManager {
```
Comment (BugenZhao) [inline src/frontend/src/session.rs:662]: May add some documentation for the special purposes and limitation of the current impl.

**Pull request merged**